homeassistant/components/meteoalarm/sensor.py

```diff
@@ -0,0 +1,99 @@
+"""Sensor for MeteoAlarm.eu."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_COUNTRY = 'country'
+CONF_PROVINCE = 'province'
+CONF_LANGUAGE = 'language'
+
+ATTRIBUTION = ("Information provided by MeteoAlarm, time delays between this"
+               "website and the www.meteoalarm.eu website are possible, for"
+               "the most up to date information about alert levels as"
+               "published by the participating National Meteorological"
+               "Services please use www.meteoalarm.eu")
+
+DEFAULT_NAME = 'meteoalarm'
+
+ICON = 'mdi:alert'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_COUNTRY): cv.string,
+    vol.Required(CONF_PROVINCE): cv.string,
+    vol.Optional(CONF_LANGUAGE, default='en'): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the MeteoAlarm sensor platform."""
+    from meteoalertapi import Meteoalert
+
+    country = config.get(CONF_COUNTRY)
+    province = config.get(CONF_PROVINCE)
+    language = config.get(CONF_LANGUAGE)
+    name = config.get(CONF_NAME)
+
+    try:
+        api = Meteoalert(country, province, language)
+    except KeyError():
+        _LOGGER.exception("Wrong country digits, or province name")
+        return
+
+    add_entities([MeteoAlertSensor(api, name)], True)
+
+
+class MeteoAlertSensor(Entity):
+    """Representation of a MeteoAlert sensor."""
+
+    def __init__(self, api, name):
+        """Initialize the MeteoAlert sensor."""
+        self._name = name
+        self._attributes = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+        }
+        self._state = None
+        self._api = api
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return ICON
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Update device state."""
+        alert = self._api.get_alert()
+        if alert:
+            for key, value in alert.items():
+                self._attributes[key] = value
+            self._state = alert['headline']
+        else:
+            self._state = 'no warnings'
```

Comment:
This looks like that it should be a binary sensor. There is a warning or there is no warning. The type of warning could be available as attribute which would make it accessible for a `template` sensor.

Reply:
Binary sensors gather information about the state of devices which have a “digital” return value (either 1 or 0). The sensor give you 'no warnings' if there is no weather warnings in your province. If there is a warning, it will display the headline text of that warning in the state field. This means it cannot be a binary sensor.

Reply:
There is a warning or there is no warning. Two states. Representing it in a human-readable form doesn't make it continuous. It doesn't matter if the window is 10 % or 95 % open, it's open. A warning for heavy rain or a thunderstorm are both states when you want to take action. Your automation rule for the documentation is showing that already. Thus, the behaviour is like a binary sensor. You go from `no_warning` to any warning. Reading, from `off` to `on`.

The content of the warning (aka the `headline`) is already an attribute as it's seems to be a free text. This makes it pretty hard to write automation rules based on the state. The available attributes to create fine-graded rule to automate things are there.

This integration will most likely be used in conjunction with the [`alert` component](https://www.home-assistant.io/components/alert/) and the simplest input there is a digital value.